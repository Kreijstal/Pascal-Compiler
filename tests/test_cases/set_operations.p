program set_operations;

type
  TDay = (Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday);
  TDaySet = set of TDay;

var
  weekdays, weekend, all_days, overlap: TDaySet;
  day: TDay;
  union_count, intersection_count: integer;
begin
  weekdays := [Monday, Tuesday, Wednesday, Thursday, Friday];
  weekend := [Saturday, Sunday];
  all_days := weekdays + weekend;
  overlap := weekdays * weekend;

  union_count := 0;
  intersection_count := 0;

  for day := Monday to Sunday do
  begin
    if day in all_days then
      union_count := union_count + 1;
    if day in overlap then
      intersection_count := intersection_count + 1;
  end;

  writeln(union_count);
  writeln(intersection_count);

  if Friday in all_days then
    writeln('weekday');

  if Saturday in all_days then
    writeln('weekend');
end.
