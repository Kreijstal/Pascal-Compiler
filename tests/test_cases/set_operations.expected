union-has-4
intersection-has-4
constructor-has-2
